pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_column.go

```diff
@@ -205,32 +204,17 @@ func dropColumn(
 				}
 			})
 		case *scpb.SecondaryIndex:
-			indexElts := b.QueryByID(e.TableID).Filter(hasIndexIDAttrFilter(e.IndexID))
-			_, _, indexName := scpb.FindIndexName(indexElts.Filter(publicTargetFilter))
-			name := tree.TableIndexName{
-				Table: *tn,
-				Index: tree.UnrestrictedName(indexName.Name),
-			}
-			dropSecondaryIndex(b, &name, behavior, e)
+			dropSecondaryIndex(b, n, behavior, e)
 		case *scpb.UniqueWithoutIndexConstraint:
-			// TODO(ajwerner): Support dropping UNIQUE WITHOUT INDEX constraints.
-			panic(errors.Wrap(scerrors.NotImplementedError(n),
-				"dropping of UNIQUE WITHOUT INDEX constraints not supported"))
+			dropConstraintByID(b, e.TableID, e.ConstraintID)
 		case *scpb.CheckConstraint:
-			// TODO(ajwerner): Support dropping CHECK constraints.
-			// We might need to extend and add check constraint to dep-rule
-			// "column constraint removed right before column reaches delete only"
-			// in addition to just `b.Drop(e)`. Read its comment for more details.
-			panic(errors.Wrap(scerrors.NotImplementedError(n),
-				"dropping of CHECK constraints not supported"))
+			dropConstraintByID(b, e.TableID, e.ConstraintID)
 		case *scpb.ForeignKeyConstraint:
-			if e.TableID != col.TableID && behavior != tree.DropCascade {
-				panic(pgerror.Newf(pgcode.DependentObjectsStillExist,
-					"cannot drop column %s because other objects depend on it", cn.Name))
+			if col.TableID != e.TableID && behavior != tree.DropCascade {
+				tn := simpleName(b, e.TableID)
+				panic(sqlerrors.NewUniqueConstraintReferencedByForeignKeyError(cn.Name, tn))
 			}
-			// TODO(ajwerner): Support dropping FOREIGN KEY constraints.
-			panic(errors.Wrap(scerrors.NotImplementedError(n),
-				"dropping of FOREIGN KEY constraints not supported"))
+			dropConstraintByID(b, e.TableID, e.ConstraintID)
```

Comment:
These changes here and above in this file should make it into the 23.1 release and be adequately tested, independently of DROP TYPE ... CASCADE.